cli/parse.go

```diff
@@ -85,6 +101,23 @@ func (st *parseState) peek() string {
 	return st.cmdline[st.i]
 }
 
+func setOpts(kv kv, opts cmds.OptMap, optDefs map[string]cmds.Option) error {
+
+	if _, exists := opts[kv.Key]; exists && optDefs[kv.Key].Type() != cmds.Strings {
+		return fmt.Errorf("multiple values for option %q", kv.Key)
+	}
+	if optDefs[kv.Key].Type() == cmds.Strings {
```

Comment:
What happens here if `optDefs[kv,Key]` is not set?

Reply:
Based on the code paths on which `setOpts` is called, this won't happen because `kv.Key` was parsed from optsDef. Since we only pull the option type from optDefs, I refactor `setOpts` accordingly.

Reply:
Great thanks